src/main/java/teammates/common/datatransfer/FeedbackRubricQuestionDetails.java

```diff
@@ -593,14 +594,74 @@ public String getQuestionResultStatisticsHtml(List<FeedbackResponseAttributes> r
                             Slots.RUBRIC_ROW_BODY_FRAGMENTS, tableBodyFragmentHtml.toString());
             tableBodyHtml.append(tableRow).append(Const.EOL);
         }
+
+        String statsTitle = "Response Summary";
+
+        if ("student".equals(view)) {
+            if (responses.size() == responsesForStatistics.size()) {
+                statsTitle = "Response Summary (of visible responses)";
+            } else {
+                statsTitle = "Response Summary (of received responses)";
+            }
+        }
         
         
         return Templates.populateTemplate(
                 FormTemplates.RUBRIC_RESULT_STATS,
-                Slots.STATS_TITLE, "student".equals(view) ? "Response Summary (of visible responses)" : "Response Summary",
+                Slots.STATS_TITLE, statsTitle,
                 Slots.TABLE_HEADER_ROW_FRAGMENT_HTML, tableHeaderFragmentHtml.toString(),
                 Slots.TABLE_BODY_HTML, tableBodyHtml.toString());
     }
+
+    /**
+     * Returns a list of FeedbackResponseAttributes filtered according to view, question recipient type
+     * for the Statistics Table
+     */
+    private List<FeedbackResponseAttributes> filterResponsesForStatistics(List<FeedbackResponseAttributes> responses,
+            FeedbackQuestionAttributes question, String studentEmail,
+            FeedbackSessionResultsBundle bundle, String view) {
+
+
+        boolean isViewedByStudent = "student".equals(view);
+        if (!isViewedByStudent) {
+            return responses;
+        }
+
+        FeedbackParticipantType[] typesToSkipFiltering = {FeedbackParticipantType.INSTRUCTORS,
+                FeedbackParticipantType.NONE, FeedbackParticipantType.SELF};
+        FeedbackParticipantType recipientType = question.getRecipientType();
+
+        for (FeedbackParticipantType skipType : typesToSkipFiltering) {
+            if (recipientType.equals(skipType)) {
+                return responses;
+            }
+        }
+
+        FeedbackParticipantType[] typesToFilterByTeams = {FeedbackParticipantType.OWN_TEAM,
+                FeedbackParticipantType.TEAMS};
+        boolean toFilterByTeams = false;
+
+        for (FeedbackParticipantType teamType : typesToFilterByTeams) {
+            if (recipientType.equals(teamType)) {
+                toFilterByTeams = true;
+            }
+        }
+
+        List<FeedbackResponseAttributes> receivedResponses = new ArrayList<>();
+        String recipientString = toFilterByTeams ? bundle.getTeamNameForEmail(studentEmail) : studentEmail;
+
+        for (FeedbackResponseAttributes response : responses) {
+            boolean isReceivedResponse = response.recipient.equals(recipientString);
+            if (isReceivedResponse) {
+                receivedResponses.add(response);
+            }
+        }
+        if (receivedResponses.isEmpty()) {
```

Comment:
@JosephineKwa @leeyimin I think we should not use one as the substitute for the other as that can be misleading (some instructors might not want students to see overall stats anyway). If no responses received, we should keep it empty or show `N/A` or something.
We should also show 'average excluding self responses' but that can be a separate issue.